ReactAndroid/src/main/java/com/facebook/react/modules/share/ShareModule.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.share;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.DialogFragment;
+import android.app.FragmentManager;
+import android.content.ComponentName;
+import android.content.DialogInterface;
+import android.content.DialogInterface.OnClickListener;
+import android.content.DialogInterface.OnDismissListener;
+import android.content.Intent;
+import android.content.pm.ResolveInfo;
+import android.net.Uri;
+import android.widget.ArrayAdapter;
+import android.os.Bundle;
+
+import com.facebook.common.logging.FLog;
+import com.facebook.react.bridge.JSApplicationIllegalArgumentException;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.WritableNativeMap;
+import com.facebook.react.common.ReactConstants;
+import com.facebook.react.common.annotations.VisibleForTesting;
+
+/**
+ * Intent module. Launch other activities or open URLs.
+ */
+public class ShareModule extends ReactContextBaseJavaModule {
+
+  @VisibleForTesting
+  public static final String FRAGMENT_TAG = "ShareAndroid";
+
+  /* package */ static final String ACTION_SHARED = "sharedAction";
+  /* package */ static final String ACTION_DISMISSED = "dismissedAction";
+
+  static final String ERROR_NO_ACTIVITY = "E_NO_ACTIVITY";
+  static final String ERROR_INVALID_CONTENT = "E_INVALID_CONTENT";
+  static final String ERROR_UNABLE_TO_OPEN_DIALOG = "E_UNABLE_TO_OPEN_DIALOG";
+  static final String ERROR_NO_PACKAGE_TO_SHARE = "E_NO_PACKAGE_TO_SHARE";
+
+  private ReactApplicationContext mContext;
+
+  public ShareModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+    mContext = reactContext;
+  }
+
+  @Override
+  public String getName() {
+    return "ShareModule";
+  }
+
+  interface OnClickItemListener {
+    void onClick(DialogInterface dialog, String packageName);
+  }
+
+  /* package */ class ShareDialogListener implements OnClickItemListener, OnDismissListener {
```

Comment:
I don't understand why this is needed. Why would the app care if the user has shared or dismissed the dialog? If it goes in other parts of the app, those parts should know if a certain action happened or not.